library/alloc/src/vec/mod.rs

```diff
@@ -3131,10 +3131,7 @@ impl<T, const N: usize> From<[T; N]> for Vec<T> {
     /// ```
     #[cfg(not(test))]
     fn from(s: [T; N]) -> Vec<T> {
-        <[T]>::into_vec(
-            #[rustc_box]
-            Box::new(s),
-        )
+        <[T]>::into_vec(Box::new(s))
```

Comment:
Same as other cases; in an unoptimized build, `T` already needs to be on the stack, so this change shouldn't inhibit any in-place heap initialization.